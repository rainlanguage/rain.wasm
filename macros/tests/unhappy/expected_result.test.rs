#[macro_use]
extern crate wasm_bindgen_utils_macros;

struct TestStruct;

#[wasm_export]
impl TestStruct {
    pub async fn some_static_method(arg: String) -> String {
        String::new()
    }
}

#[wasm_export]
impl TestStruct {
    pub async fn some_static_method() {}
}

fn main() {}