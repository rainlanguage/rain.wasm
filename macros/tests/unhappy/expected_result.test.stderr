error: expected Result<T, E> return type
 --> tests/unhappy/expected_result.test.rs:8:50
  |
8 |     pub async fn some_static_method(arg: String) -> String {
  |                                                  ^^^^^^^^^

error: expected Result<T, E> return type
  --> tests/unhappy/expected_result.test.rs:15:9
   |
15 |     pub async fn some_static_method() {}
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
